electronics_lib/JLC_Resistor.py

```diff
@@ -0,0 +1,87 @@
+class JLCPCB_ResistorTable(ESeriesResistor):
+  RESISTANCE = PartsTableColumn(Range)  
+  POWER_DISSIPATION = PartsTableColumn(Range)  
+  PRICE = PartsTableColumn(float)
+  FOOTPRINT = PartsTableColumn(str)
+    
+  @classmethod
+  def _generate_table(cls) -> PartsTable:
+    RESISTOR_MATCHES = {
+    'resistance': "\s?(\d+(\.\d*)?[GMkmunp]?)?[\u03A9]\s",
+    'tolerance': "\s?([\u00B1]\d+(\.\d*)?%)\s",
+    'power_dissipation': "\s?([\u00B1]\d+(\.\d*)?ppm)/℃\s"
+    }
+
+    def parse_row(row: PartsTableRow) -> Optional[Dict[PartsTableColumn, Any]]:
+      if (row['Library Type'] != 'Basic'):
+          return None            
+      if (row['First Category'] != 'Resistors'):
+          return None
+
+      new_rows: Dict[PartsTableColumn, Any] = {}
+      try:
+      # handle the footprint first since this is the most likely to filter
+      footprint = ""
+      new_rows[cls.FOOTPRINT] = footprint
+    
+      extracted_values = parse(row['Description'], RESISTOR_MATCHES)
+    
+      new_rows[cls.RESISTANCE] = Range.from_tolerance(
+          PartsTableUtil.parse_value(extracted_values['resistance'][0], 'Ohm'),
+          PartsTableUtil.parse_tolerance(extracted_values['tolerance'][0])
+      )
+      new_rows[cls.POWER_DISSIPATION] = PartsTableUtil.parse_value(extracted_values['power_dissipation'][0])
+      new_rows[cls.PRICE] = float(max(re.findall(":(\d+\.\d*),", row['Price'])))
```

Comment:
I could see this being refactored into a JlcPcbTable class, since other parts will also use the same cost parser.